test/hotCases/invalidate/during-idle/index.js

```diff
@@ -11,9 +11,10 @@ it("should allow to invalidate and reload a file", () => {
 	expect(module.hot.status()).toBe("ready");
 	c.invalidate();
 	expect(module.hot.status()).toBe("ready");
-	module.hot.apply();
-	expect(module.hot.status()).toBe("idle");
-	expect(a.value).not.toBe(oldA);
-	expect(b.value).not.toBe(oldB);
-	expect(c.value).toBe(oldC);
+	module.hot.apply().then(function () {
```

Comment:
Make sure that the `apply()` API stay sync as long there is no async status handler registered, to avoid a breaking change.

Since `setStatus` is only internal you could return a `{ then: fn => fn() }` when no Promise was returned from status handlers

Reply:
Depending on the synchronously-executed part of an API that's documented to return a `Promise` seems dangerous, and even before this patch error handling is broken if you try to do so. But I can still implement the synchronous `then()` if you'd like.